src/Effect/Ref.purs

```diff
@@ -15,6 +15,10 @@ foreign import data Ref :: Type -> Type
 -- | Create a new mutable reference containing the specified value.
 foreign import new :: forall s. s -> Effect (Ref s)
 
+-- | Create a new mutable reference containing a value that can refer to the
+-- | `Ref` being created.
+foreign import self :: forall s. (Ref s -> s) -> Effect (Ref s)
```

Comment:
What about naming this function `newSelf` instead to highlight that it is a variant of the `new` function? To me, the name `self` and nothing else seems a bit vague. If I saw the name `newSelf` I would immediately know that the function is related to `new`.